base-notebook/start.sh

```diff
@@ -16,6 +16,9 @@ if [ $UID == 0 ] ; then
                 chown -R $NB_UID "$d"
             fi
         done
+    elif [[ -d /home/$NB_USER/work && $(stat -c %U /home/$NB_USER/work) != "$NB_USER" ]]; then
```

Comment:
In kubernetes, you can set fsGroup in Security Context to 1000 and solve this problem :) Look at z2jh.jupyter.org for a nice k8s setup!